### PR TITLE
feat: add random unrandomizer

### PR DESCRIPTION
**Pull request by chown2**
Simple search bar based interface, while we are figuring out how to do add a GUI for this.

### CODE REVIEW

**Comment by MatVeiQaaa** [comment]
why not p2 arrange display too?

File: LR2/Scene02_Songselect.cpp [hunk 1]
```diff
@@ -1283,6 +1283,35 @@ int CmdSearch(game *g, CSTR *cmd, sqlite3 *sql) {
...
+		if (ec != std::errc()) {
+			*cmd = ("INVALID NUMBER: " + std::string(s)).c_str();
+			return 1;
+		}
+		if (val == 0) {
+			g->gameplay.forceRandomLayout = 0;
+			*cmd = g->gameplay.forceRandomLayout == 0 ? "OFF" : std::to_string(g->gameplay.forceRandomLayout).c_str();
+		}
+		else if (val >= 12345 && val <= 54321) {
+			g->gameplay.forceRandomLayout = val * 100 + 67;
```
Comment (MatVeiQaaa) [inline LR2/Scene02_Songselect.cpp:1303]: LOL genius

**Review by MatVeiQaaa — APPROVED** [review]
it's alright, a bit of a mess in seed acquisition bit, but it's a short piece of code, whatever.

**Comment by MatVeiQaaa** [comment]
i think you should add p2 arrange display though

**Comment by chown2** [comment]
~~14k doesn't exist~~ compared to P1 I don't have a skin to test with, and I don't know if anyone would actually use that to justify spending 5 minutes to add random display into a new skin. :D Do note that the DST number for P2, 418, is only used for DP.

**Comment by Sonic-TV** [comment]
> ~~14k doesn't exist~~ compared to P1 I don't have a skin to test with, and I don't know if anyone would actually use that to justify spending 5 minutes to add random display into a new skin. :D Do note that the DST number for P2, 418, is only used for DP.

I've used plenty of LR2OOL exclusive stuff in my skin `extended-tricoro-for-lr2`, you may test with that.

**Comment by chown2** [comment]
@Sonic-TV thank you, it helps, I will add P2 display. Do note I do not endorse skins using extracted textures however. :)

**Comment by Sonic-TV** [comment]
> @Sonic-TV thank you, it helps, I will add P2 display. Do note I do not endorse skins using extracted textures however. :)

I was thinking of making something better than LITONE and not let everyone paying for this, so yeah (lol).

**Comment by chown2** [comment]
Added P2 random layout display. @Sonic-TV your skin uses the numbers in a different way too, which exposed an incorrectly handled edge case. Thank you!

**Comment by Sonic-TV** [comment]
> Added P2 random layout display. @Sonic-TV your skin uses the numbers in a different way too, which exposed an incorrectly handled edge case. Thank you!

I've used all of the button and text definitions in skin select (and plenty of `a/*/c` structures in entire skin as well), so you may test the functionality with it.

**Review by GOMazk — APPROVED** [review]
what a magic show lol
